frontend/apps/reader/readerui.lua

```diff
@@ -871,4 +871,11 @@ function ReaderUI:getCurrentPage()
     end
 end
 
+function ReaderUI:onMenuSearch()
+    if not self.ui then
+        UIManager:sendEvent(Event:new("ShowMenu"))
+    end
+    UIManager:broadcastEvent(Event:new("MenuSearchDialog"))
```

Comment:
Not sure why this.
If `not self.ui`, there is nothing that will catch your ShowMenu event.
So, the first sendEvent could be just `self.ui.menu:onShowMenu()`.
But why is it needed ? Because you need to get once the menu opened so it's populated? Or else ?
(Here and in FileManager:onMenuSearch.)

Also, looking at the event names in Dispatcher, these should probably all be named on**Show**MenuSearch...

Reply:
Calling `self.ui.menu:onShowMenu()` would be the first joice, but if only the reader is open, there is no `self.ui`. That's why I open the menu with an event (in reader and fm).

+1 to `onShowMenuSearch`

Reply:
> Calling self.ui.menu:onShowMenu() would be the first joice, but if only the reader is open, there is no self.ui. That's why I open the menu with an event (in reader and fm).

I don't get that.
But may be it's because you moved that to ReaderUI, where there is no and never will be any `self.ui` - because it is itself (ReaderUI) what is known as self.ui for all the reader modules.
So, it may be just `self.menu:onShowMenu()` - and `self.menu.handleEvent(Event:new("MenuSearchDialog"))`

Reply:
Oh, thanks.I will look into this at WE.